fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/MessageHandler.java

```diff
@@ -37,45 +197,23 @@ private MessageHandler() {
    * @param channel the channel on which the message was received
    * @param message the message that was received
    */
-  public static void handleMessage(
-      LAORepository laoRepository, String channel, MessageGeneral message)
+  public void handleMessage(LAORepository laoRepository, String channel, MessageGeneral message)
       throws DataHandlingException {
     Log.d(TAG, "handle incoming message");
 
     // Put the message in the state
     laoRepository.getMessageById().put(message.getMessageId(), message);
 
-    String senderPk = message.getSender();
-
     Data data = message.getData();
     Log.d(TAG, "data with class: " + data.getClass());
 
     Objects dataObj = Objects.find(data.getObject());
     if (dataObj == null) throw new UnknownDataObjectException(data);
 
-    switch (dataObj) {
-      case LAO:
-        handleLaoMessage(laoRepository, channel, data, message.getMessageId());
-        laoRepository.updateNodes(channel);
-        break;
-      case ROLL_CALL:
-        handleRollCallMessage(laoRepository, channel, data, message.getMessageId());
-        break;
-      case ELECTION:
-        handleElectionMessage(laoRepository, channel, data, message.getMessageId(), senderPk);
-        break;
-      case CONSENSUS:
-        handleConsensusMessage(laoRepository, channel, data, message.getMessageId(), senderPk);
-        break;
-      case CHIRP:
-        handleChirpMessage(laoRepository, channel, data, message.getMessageId(), senderPk);
-        break;
-      case MESSAGE:
-        handleWitnessMessage(laoRepository, channel, senderPk, data);
-        break;
-      default:
-        throw new UnhandledDataTypeException(data, dataObj.getObject());
-    }
+    Action dataAction = Action.find(data.getAction());
+    if (dataAction == null) throw new UnknownDataActionException(data);
```

Comment:
Actually, this should never occur, as the JsonDeserialization will fail if the pair is not registered. 
So the packet execution will not go to this point.